code/modules/events/anomaly_cola.dm

```diff
@@ -0,0 +1,54 @@
+#define COLA_AMT 5
+
+/datum/round_event_control/infcola
+	name = "Cola Infinite"
+	typepath = /datum/round_event/infcola
+	min_players = 1
+	max_occurrences = 10
+	var/atom/special_target
+	can_malf_fake_alert = TRUE
+
+/datum/round_event/infcola
+	announceWhen = 2
+
+/datum/round_event/infcola/announce(fake)
+	priority_announce("Our long-range anomaly scanners have detected leakage from a soda filled dimension. Nanotrasen is [fake ? "very" : "not"] responsible for any damages caused by these anomalous canisters.", "General Alert", SSstation.announcer.get_rand_alert_sound())
+
+/obj/item/reagent_containers/food/drinks/soda_cans/inf
+	name = "Space Cola INFINITE"
+	desc = "Cola. Probably not from space. Proceed with caution. A no-tamper filter prevents the drink from being drained and resold."
+	icon_state = "inf"
+	list_reagents = list(/datum/reagent/consumable/space_cola = COLA_AMT)
+	prevent_grinding = TRUE
+	reagent_flags = AMOUNT_VISIBLE | REFILLABLE  // disables dumping chems into container, you can still fill with dangerous chems for a cool trick
+
+/obj/item/reagent_containers/food/drinks/soda_cans/inf/Initialize(mapload)
+	var/reagents = volume - COLA_AMT
```

Comment:
isn't reagents already a var on the object? Could get confusing if you override the variable locally when reagents already exists somewhere else

Reply:
reagents_left would be more suitable.